pkg/build/builder/common.go

```diff
@@ -180,3 +204,23 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface) error {
+	return kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
+		// before updating, make sure we are using the latest version of the build
+		latestBuild, err := client.Get(build.Name)
```

Comment:
adding the get rule to the system:image-builder role would work:
```
authorizationapi.NewRule("get").Groups(buildGroup).Resources("builds").RuleOrDie(),
```
you need to run `oadm policy reconcile-cluster-roles --confirm` after rebuilding to update the loaded policy in your cluster